lib/migration/manager/manager.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/migration/errors"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/version"
+	"github.com/vmware/vic/pkg/vsphere/session"
+)
+
+const (
+	ApplianceConfigure = "ApplianceConfigure"
+	ContainerConfigure = "ContainerConfigure"
+
+	ApplianceVersionKey = "guestinfo.vice./init/version/PluginVersion"
```

Comment:
From the impl point of view, it's right to leverage tools to calculate, in case they might change. 
But here, I think we actually rely on one constant, which should not change in all our versions.
This hardcoded string makes me believe that's really not changed, how do you think?